Add Mongo storage solution

filipecabaco opened the pull request:
First step to replace Cassandra as our main entry storage.

Commit:
Add Mongo storage solution

First step to replace Cassandra as our main entry storage.

jvvasques commented:
Looks good to me. A docker image would be nice to have 👍  what do you say @nmcc ?

filipecabaco commented:
already doing that, almost there fully replacing cassandra but I'm having problem with _id's repeated. 

I need to recheck what I'm doing but it should be quick 👍

nmcc commented on talkpocket-api/src/talkpocket_api/storage/mongo.clj:
```diff
@@ -0,0 +1,43 @@
+  ([id] (keywordize (coll/find-map-by-id @db collection (ObjectId. id)))))
+
+(defn consumer
+  "Consumer that receives a map and persists it to Cassandra"
```
`persists it to Cassandra`?? 🤔

filipecabaco replied:
still doing those changes, this was just to showcase mongo in clojure

jvvasques replied:
:trollface:

nmcc reviewed (commented):
Quite simple change, indeed

nmcc commented:
Docker FTW!!

filipecabaco commented:
I'm actually tempted to remove docker stuff and use startup scripts... less prone to error.

Just keep the services dockerized

Commit:
Replace Cassandra with Mongo

Completely removed Cassandra and replaced it with Mongo.

Get File still has problems but at least we are able to create an
entry at the beginning.

Baby steps... We're improving!

filipecabaco commented:
Still has problems with minio, almost there solving that one, after that I'm going to check for a better way to generate id's that follow the idea of URL and language as a defining element of ID generation but also following a similar idea to https://blog.twitter.com/2010/announcing-snowflake

Commit:
Fixed all issues

Finally fixed all the issues with using Mongo instead of Cassandra.

All operations where tested and they return the expected values,
we were also able to use the base64 encoded id to save on Mongo which
means we retained all the desired features.

:metal:
